vortex-datafusion/src/persistent/format.rs

```diff
@@ -153,24 +153,31 @@ impl FileFormat for VortexFormat {
 
     async fn infer_schema(
         &self,
-        _state: &SessionState,
+        state: &SessionState,
         store: &Arc<dyn ObjectStore>,
         objects: &[ObjectMeta],
     ) -> DFResult<SchemaRef> {
-        let file_schemas = stream::iter(objects.iter().cloned())
+        let mut file_schemas = stream::iter(objects.iter().cloned())
             .map(|o| {
                 let store = store.clone();
                 let cache = self.file_layout_cache.clone();
                 async move {
                     let file_layout = cache.try_get(&o, store).await?;
                     let inferred_schema = infer_schema(file_layout.dtype())?;
-                    VortexResult::Ok(inferred_schema)
+                    VortexResult::Ok((o.location, inferred_schema))
                 }
             })
-            .buffered(self.opts.concurrent_infer_schema_ops)
+            .buffer_unordered(state.config_options().execution.meta_fetch_concurrency)
```

Comment:
unleashing the kraken

Reply:
I also looked into using a `JoinSet` here but the error handling is annoying and I assume that for the vast majority of use cases it won't make much difference.